### ISSUE TITLE
Error in Search-SendOnBehalfPermission

### ISSUE THREAD

**Issue by astavitsky** [JBines/Get-RecipientPermissions.ps1#4]
`Search-SendOnBehalfPermission` returns an error when SendOnBehalfPermissions are found:

```
Get-RecipientPermissions.ps1 : Failed to call function Search-SendOnBehalfPermission: Cannot bind argument to parameter 'User' because 
it is null..Exception.Message
At line:1 char:1
+ Get-RecipientPermissions.ps1 -Identity $user4
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,Get-RecipientPermissions.ps1
```

It appears that inside `Search-SendOnBehalfPermission` function `$SOBPobj` is a simple canonical DN and not a user object.  Yet it is used as if it is a user object on lines 953, 960, 986

Changing `$SOBPobj.DistinguishedName` to `$SOBPObj` on line 953 and `$SOBPObj` to `$SOBPobj_USER` on lines 960 and 986 seems to solve the issue.

**Comment by astavitsky** [comment]
It looks like the issue is only present when running using remote PowerShell Exchange session.  It works correctly when run in Exchange Management shell on the Exchange server itself.
Apparently, the Exchange Management shell is loading the Exchange assemblies prior to initiating remote session and the `Get-Mailbox` return attributes have different type binding in Exchange Management shell.

I suggest that you either close this and reinforce the requirement to run this script from Exchange Management shell or update the script in multiple places (not just `Search-SendOnBehalfPermission) so it works in both Exchange Management shell and in regular remote Exchange session.

**Comment by JBines** [comment]
Sorry for the late reply. 

I was hoping I could spend the time to update and test but it seems time is not my friend :( is it ever!

I changed the order of the notes and added IMPORTANT! hopefully that helps. 

.NOTES
IMPORTANT! We recommend you run this script from the Exchange Management Shell for the best results. 

I also added to the ever growning todo list... 

 HIGH - Add Support for EXO
 HIGH - Add Support Remote PS Sessions
